### PR TITLE
(FM-7470) change method names, use upload_file

### PR DESCRIPTION
**Pull request by tphoney**
This PR is to try and utilise more of boltspec in bolt and to better name some of the helper functions.

### CODE REVIEW

File: lib/puppet_litmus/serverspec.rb [hunk 1]
```diff
@@ -51,10 +51,9 @@ def create_manifest_file(manifest)
...
       # transfer to TARGET_HOST
-      command = "bundle exec bolt file upload #{manifest_file.path} /tmp/#{File.basename(manifest_file)} --nodes #{target_node_name} --inventoryfile inventory.yaml"
-      stdout, stderr, status = Open3.capture3(command)
-      error_message = "Attempted to run\ncommand:'#{command}'\nstdout:#{stdout}\nstderr:#{stderr}"
-      raise error_message unless status.to_i.zero?
+      inventory_hash = inventory_hash_from_inventory_file
+      result = upload_file(manifest_file.path, "/tmp/#{File.basename(manifest_file)}", target_node_name, options: {}, config: nil, inventory: inventory_hash)
+      raise result.first['result'].to_s unless result.first['status'] == 'success'
 
       manifest_file_location = "/tmp/#{File.basename(manifest_file)}"
```
Comment (ekohl) [inline lib/puppet_litmus/serverspec.rb]: Probably nicer to define this variable beforehand and then use it as an argument in `upload_file`

Comment (tphoney) [reply lib/puppet_litmus/serverspec.rb]: are you talking about `manifest_file_location` ?

File: lib/puppet_litmus/serverspec.rb [hunk 2]
```diff
@@ -51,10 +51,9 @@ def create_manifest_file(manifest)
       manifest_file_location = manifest_file.path
     else
       # transfer to TARGET_HOST
-      command = "bundle exec bolt file upload #{manifest_file.path} /tmp/#{File.basename(manifest_file)} --nodes #{target_node_name} --inventoryfile inventory.yaml"
-      stdout, stderr, status = Open3.capture3(command)
-      error_message = "Attempted to run\ncommand:'#{command}'\nstdout:#{stdout}\nstderr:#{stderr}"
-      raise error_message unless status.to_i.zero?
+      inventory_hash = inventory_hash_from_inventory_file
+      result = upload_file(manifest_file.path, "/tmp/#{File.basename(manifest_file)}", target_node_name, options: {}, config: nil, inventory: inventory_hash)
```
Comment (ekohl) [inline lib/puppet_litmus/serverspec.rb]:
```suggestion
      result = upload_file(manifest_file.path, manifest_file_location, target_node_name, options: {}, config: nil, inventory: inventory_hash)
```

File: lib/puppet_litmus/serverspec.rb (hunk 1 above)
Comment (ekohl) [reply lib/puppet_litmus/serverspec.rb]: Yes. So define it before or after `inventory_hash`.

**Comment by tphoney** [comment]
An experiment was done to use apply_manifest from boltspec, but it was significantly slower. https://github.com/tphoney/puppet_litmus/tree/bolt_apply_manifest and meant that apply_manifest would be renamed due to Ruby not handling overloading.

**Pull request merged**